hibernate-core/src/main/java/org/hibernate/engine/transaction/internal/TransactionImpl.java

```diff
@@ -37,12 +37,20 @@
 	public TransactionImpl(
 			TransactionCoordinator transactionCoordinator,
 			ExceptionConverter exceptionConverter,
-			JpaCompliance jpaCompliance) {
+			JpaCompliance jpaCompliance,
+			boolean sessionIsClosed) {
```

Comment:
I wonder if I got too cute here.  I wonder if we should just pass (some form of) the Session into the Transaction here.  Then we could just ask the Session if it is open.

I did it this way for "testability", but I'm not sure that goal worth not having the Session available here.